app/main.py

```diff
@@ -1,3 +1,69 @@
+from typing import Callable, Any
+
+
+def validate_operand(func: Callable) -> Callable:
+    def wrapper(self: "Distance", other: ("Distance", int, float)) -> Any:
+        if isinstance(other, Distance):
+            return func(self, other.km)
+        elif isinstance(other, (int, float)):
+            return func(self, other)
+        else:
+            raise TypeError(f"Unsupported operand type for {func.__name__}")
+
+    return wrapper
```

Comment:
GJ)

Reply:
but add a functool.wraps for inner function as a best practice

Reply:
Done